morecantile/models.py

```diff
@@ -1367,20 +1380,30 @@ def feature(
             },
         }
 
-        if projected:
+        if feature_crs != WGS84_CRS:
             warnings.warn(
                 "CRS is no longer part of the GeoJSON specification."
                 "Other projection than EPSG:4326 might not be supported.",
                 UserWarning,
             )
-            feat.update(
-                {
-                    "crs": {
-                        "type": "EPSG",
-                        "properties": {"code": self.crs.to_epsg()},
+            if epsg := feature_crs.to_epsg():
```

Comment:
probably could condense this using `to_authority` to avoid the EPSG switch following the logic in https://github.com/cogeotiff/rio-tiler/blob/dd5ce03f8f463ec93f3a8787d7155d318397cd7c/rio_tiler/utils.py#L808-L823

Reply:
thinking this would be an actually pretty important improvement

Reply:
I'm not sure to get this @AndrewAnnex 

are you saying that `to_authority` is better than `to_epsg` in term of performance, or do you mean we should use `to_authoriy` and save the CRS info like 
```python
    if authority_code := feature_crs.to_authority(min_confidence=20):
        authority, code = authority_code
        feat.update(
            {
                "crs": {
                    "type": authority,
                    "properties": {"code": code},
                }
            }
        )
    else:
        feat.update(
            {
                "crs": {
                    "type": "name",
                    "properties": {"name": CRS_to_uri(feature_crs)},
                }
            }
        )
```

Reply:
yeah something like the above code snippet would be what I was thinking. `to_epsg` isn't necessarily slow, but effectively enforcing that the only proper CRS authority is the epsg is what I was wanting to avoid.

It could be effective to just only ever use the CRS_to_uri option here instead of splitting the behavior, I think if an authority is not identified the URI wouldn't be usable anyways (the function would return None maybe?).

Reply:
what do you think about 

```python
if authority_code := feature_crs.to_authority(min_confidence=20):
    authority, code = authority_code
    feat.update(
        {
            "crs": {
                "type": authority,
                "properties": {"code": code},
            }
        }
    )
else:
    feat.update(
        {
            "crs": {
                "type": "wkt",
                "properties": {"wkt": feature_crs.to_wkt()},
            }
        }
    )
```

Because there is no specification for CRS we can pretty much do what we want 😅

Reply:
the `original` spec was 
```
  "crs": {
    "type": "name",
    "properties": {
      "name": "urn:ogc:def:crs:OGC:1.3:CRS84"
      }
    }
```
https://gist.github.com/sgillies/1233327#file-geojson-spec-1-0-L256-L271

Reply:
yeah I think having a WKT last option fallback is good. It might just be better to swap in the URI for the default rather than the code, as if you are able to resolve an authority to get a code, I suspect the URI would also then resolve. So that would mean the default behavior would be more like the original spec, and if no authority is found go to the wkt, but truly your most recent suggestion is fine.